internal/proxyservice/http/conjur/connector.go

```diff
@@ -0,0 +1,51 @@
+package conjur
+
+import (
+	"encoding/base64"
+	"fmt"
+	gohttp "net/http"
+	"strconv"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/log"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+)
+
+// Connector injects an HTTP request with Cponjur's authorization header(s).
+type Connector struct {
+	logger log.Logger
+}
+
+// Connect is the function that implements the http.Connector func
+// signature. It has access to the client http.Request and the credentials (as a
+// map), and is expected to decorate the request with Authorization headers.
+//
+// Connect uses the "accessKeyId", "secretAccessKey" and optional "accessToken"
+// credentials to sign the Authorization header, following the AWS signature
+// format.
+func (c *Connector) Connect(
+	req *gohttp.Request,
+	credentialsByID connector.CredentialValuesByID,
+) error {
+	var ok bool
+
+	accessToken, ok := credentialsByID["accessToken"]
+	if !ok {
+		return fmt.Errorf("Conjur credential 'accessToken' is not available")
+	}
+
+	forceSSLStr, ok := credentialsByID["forceSSL"]
+	forceSSL, err := strconv.ParseBool(string(forceSSLStr))
+	if ok && err == nil && forceSSL {
```

Comment:
Perhaps work for later but this should probably become a generic http proxy service option

Reply:
Yup - probably for later